evalml/automl/automl_algorithm/default_algorithm.py

```diff
@@ -500,6 +500,18 @@ def _transform_parameters(self, pipeline, proposed_parameters):
                 and self._batch_number > 0
             ):
                 component_parameters["columns"] = self._pipeline_params[name]["columns"]
+            # Check that Drop Row Transformer is before estimator
+            if (
+                is_time_series(self.problem_type)
+                and name == "Drop Rows Transformer"
+                and pipeline.component_graph.compute_order[-2] == name
+            ):
+                n_rows_to_drop = (
+                    self._pipeline_params["pipeline"]["max_delay"]
```

Comment:
@christopherbunn not all of the pipelines have the same number of NaN rows. After the first batch, the number of delays, rolling window size will change as the search progresses. 

```python
from evalml.automl import AutoMLSearch
from evalml.demos import load_weather

X, y = load_weather()

automl = AutoMLSearch(X, y, "time series regression",
                      max_batches=4, _automl_algorithm="iterative",
                      problem_configuration={"max_delay": 30, "forecast_horizon": 7, "gap": 1,
                                             "time_index": "Date"},
                      verbose=True)

automl.search()

light_gbm_pipelines = [automl.get_pipeline(i) for 
                       i in automl.full_rankings.loc[automl.full_rankings.pipeline_name.str.contains("LightGBM")].id]

[pl.parameters['Time Series Featurizer']['conf_level'] for pl in light_gbm_pipelines]


[pl.parameters['Time Series Featurizer']['rolling_window_size'] for pl in light_gbm_pipelines]

fitted_light_gbm_pipelines = [pl.fit(X, y) for pl in light_gbm_pipelines]

[pl.transform_all_but_final(X, y).isna().sum(axis=0).max() for pl in fitted_light_gbm_pipelines]
```

<img width="855" alt="image" src="https://user-images.githubusercontent.com/41651716/153646508-1ebc9d14-5232-49f6-9491-737c52028c6e.png">

<img width="873" alt="image" src="https://user-images.githubusercontent.com/41651716/153646576-60a5f621-e06e-4f6d-bbc2-3fbfbdf9006c.png">

Note that the number of NaN rows is never `max_delay` + `forecast_horizon` + `gap`. I worry we may be throwing out too much data with the approach. I like the idea of only dropping NaNs for some estimators. Can we test with DropNaN component for the estimators that need it for more than two batches?

Reply:
Sure! Here's the results I got for first N vs. drop NaN only for select estimators. It looks like across the board we're getting better results with the drop NaN component than the drop rows component. As you pointed out, it's most likely because we're dropping rows that do not have NaN values. I'll adjust the preprocessing chain to use the drop NaN component for time series pipelines when needed. 

[drop_n_vs_drop_nan_necessary.html.zip](https://github.com/alteryx/evalml/files/8062520/drop_n_vs_drop_nan_necessary.html.zip)